client/state/stored-cards/reducer.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	STORED_CARDS_FETCH,
+	STORED_CARDS_FETCH_COMPLETED,
+	STORED_CARDS_FETCH_FAILED,
+	STORED_CARDS_DELETE,
+	STORED_CARDS_DELETE_COMPLETED,
+	STORED_CARDS_DELETE_FAILED
+} from 'state/action-types';
+
+import { createStoredCardsArray } from './assembler.js';
+
+/**
+ * List all known stored cards of the current user at /me/stored-cards.
+ *
+ * @param  {Array}  state  Current state
+ * @param  {Object} action storeCard action
+ * @return {Array}         Updated state
+ */
+export const items = ( state = [], action ) => {
+	switch ( action.type ) {
+		case STORED_CARDS_FETCH:
+		case STORED_CARDS_FETCH_FAILED:
+			return [];
```

Comment:
True. The original reducer wasn't clearing the list, we should do the same here.